saleor/graphql/product/mutations/products.py

```diff
@@ -1784,6 +1784,67 @@ def perform_mutation(cls, _root, info, product_id, images_ids):
         return ProductImageReorder(product=product, images=images)
 
 
+class ProductVariantReorder(BaseMutation):
+    product = graphene.Field(Product)
+    variants = graphene.List(ProductVariant)
+
+    class Arguments:
+        product_id = graphene.ID(
+            required=True,
+            description="Id of product that variants order will be altered.",
+        )
+        variants = graphene.List(
+            graphene.ID,
+            required=True,
+            description="IDs of a product variants in the desired order.",
+        )
+
+    class Meta:
+        description = "Changes ordering of the product variants."
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = ProductError
+        error_type_field = "product_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, product_id, variants):
+        product = cls.get_node_or_error(
+            info, product_id, field="product_id", only_type=Product
+        )
+
+        if len(variants) != product.variants.count():
+            raise ValidationError(
+                {
+                    "order": ValidationError(
```

Comment:
Shouldn't `variants` here? The dashboard needs to know the field name to assign the error

Reply:
👮